src/Engine/ProtoCore/ProcedureTable.cs

```diff
@@ -392,30 +373,92 @@ public int IndexOf(string name, List<ProtoCore.Type> argTypeList, bool isStaticO
         /// <param name="functionName"></param>
         /// <param name="parameterTypes"></param>
         /// <returns></returns>
+        [Obsolete("This method will be removed in Dynamo 3.0. Please use GetFunctionBySignature with ProcedureMatchOptions instead.")]
         public ProcedureNode GetFunctionBySignature(string functionName, List<Type> parameterTypes)
         {
-            foreach (var f in Procedures)
+            GetFunctionBySignature(new ProcedureMatchOptions() 
+            { 
+                FunctionName = functionName, 
+                ParameterTypes = parameterTypes 
+            }, out ProcedureNode output);
+            return output;
+        }
+
+        /// <summary>
+        /// Get function by its signature.
+        /// </summary>
+        /// <param name="opts">Matching options</param>
+        /// <param name="outputProcNode">Output procedure node. Null if nothing is found</param>
+        /// <returns>Index of the ProcedureNode in the Procedures list. Returns -1 If nothing is found</returns>
+        public int GetFunctionBySignature(ProcedureMatchOptions opts, out ProcedureNode outputProcNode)
```

Comment:
Can we just make this internal?

Reply:
If we make this internal (which I think I am ok with) - then we should update the above obsolete message to say it's no longer supported or something and to request it if needed maybe.

Reply:
I made them internal